src/citrine/_serialization/properties.py

```diff
@@ -340,8 +357,17 @@ def _serialize(self, value: list) -> list:
             serialized.append(self.element_type.serialize(element))
         return serialized
 
+    def _set_elements(self, value):
+        elems = []
+        for sub_val in value:
+            if issubclass(type(sub_val), self.element_type.underlying_types):
```

Comment:
I'm not so sure about this if statement

Reply:
This seems reasonable, but I'm also a bit worried about the `element_type: typing.Union[Property, typing.Type[Property]]`.  Do we have test coverage of both parts of that union?

Reply:
I just tried:
```
class DummyDescriptor(Serializable):
    """docstring for dummyDescriptor"""
    dummy_map = properties.Mapping(properties.Float, properties.String, "foo")
    dummy_set = properties.Set(type(properties.Float), "bar")
```
which led to errors.  Maybe the typing bit of those unions is cruft that should be removed.

Reply:
I don't think we're expecting to handle `properties.Set(type(properties.Float), "bar")`. The three things I can think of us expecting to handle is: 
`properties.Set(type(properties.Float()), "bar")`
`properties.Set(properties.Float, "bar")`
`properties.Set(properties.Float(), "bar")`
And the last two are in use in the code base

I'll extend the test to handle all three, but the code as is handles it, in particular because of `self.element_type = element_type if isinstance(element_type, Property) else element_type()`